src/gui/qgsattributeform.cpp

```diff
@@ -3156,7 +3154,8 @@ void QgsAttributeForm::updateFieldDependenciesDefaultValue( QgsEditorWidgetWrapp
   const QSet<QString> referencedColumns = exp.referencedColumns();
   for ( const QString &referencedColumn : referencedColumns )
   {
-    if ( referencedColumn == QgsFeatureRequest::ALL_ATTRIBUTES )
+    //concern all attributes if a function (perhaps) requires all attributes or is a now-function
+    if ( referencedColumn == QgsFeatureRequest::ALL_ATTRIBUTES || exp.referencedFunctions().contains( QStringLiteral( "now" ) ) )
```

Comment:
This feels a bit ad-hoc, and won't scale up if other functions have similar requirements. I'm thinking for e.g. uuid(), rand(), randf(). 

Wondering if that could all be regrouped into maybe a static QgsExpressionUtils function where such a list could be maintained.

Reply:
Ideally we would flag these functions, similar to [`isStatic()`](https://github.com/qgis/QGIS/blob/2e7c0e6b32e0d39f72463ae4651b42109e0a2de3/src/core/expression/qgsexpressionfunction.h#L244), (e.g. `isVolatile()`) instead of maintaining a list. If someone registers a python based function they should also be able to make it behave this way.
Ideally `referencedFunctions()` (or a sibling method) would already return a list of `QgsExpressionFunction` but since this is not the case a helper function in `QgsExpressionUtils` that proxies to this can be an easier option.

Reply:
Consider that this is about the "live" update of the widget value in the form and not if it updates at all. And neither it has something go do with everything besides the default-value-on-update (like constraints etc). So I'm not sure if a flag is the right thing and neither if I would have it in the ExpressionUtils. In my opinion the attribute form is the better place...

Reply:
But I agree to have a static function for it...

Reply:
In this case, we could keep it accessible via python API of the feature form (in this case it would be a non static function of the form to get and set the list of volatile function names)?

Reply:
And where it should be set? On every attributeform call?

Reply:
I would initialize in the constructor (but leave it to python code to override)

Reply:
Hmm... That sounds reasonable, but I'm a little unsure. As I mentioned in this PR, this fixes some bad behavior here, but it may not be the final solution. Maybe every feature should be considered and it should be controlled by a user setting whether this specific widget should be live-updated or not. So I'm a little afraid that with your suggestion we are introducing something that we will soon have to mark as deprecated....